### DISCUSSION TITLE
Complex Queries using supabase_flutter

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#12025] [Questions] [answered]
I'm trying to understand how to do complex Queries within a flutter app using supabase_flutter. 
As an example, Is this possible somehow: 
Select * from claims where (status = 'open' and (excess > 4000000 or group > 500000)) and (claim = '12345' or claim = '56789')
I know this works in SQL, but I want to use these types of complex queries in the flutter app. 
The above is a sample, I'll have others, just trying to see how this would be possible without using rpc.

**Comment by GaryAustin1** [comment, marked as answer]
Assuming the flutter code works similar to the js code then you should be able to do those.

First all filters are anded together:  ` .eq('status','open').gt('excess',400000)`  (you have to meet both filters)
Then there is an `or` filter https://supabase.com/docs/reference/dart/or  (note from that link you can see all the other filters).

So you can do things like:  `.eq('status','open').or('excess.gt.4000000, group.gt.50000).or('claim.eq.12345, claim.eq.56555)`
Note on the last `or` you could use `in` instead. ` .in('claim',['12345','56769'])`

I've not checked the syntax against flutter (in particular the array) but it should be similar.
